app/questionnaire/navigator.py

```diff
@@ -77,29 +76,24 @@ def build_path(self, blocks, this_location):
             # If routing rules exist then a rule must match (i.e. default goto)
             if 'routing_rules' in block and len(block['routing_rules']) > 0:
                 original_this_location = this_location.copy()
-                for rule in block['routing_rules']:
-                    if SchemaHelper.is_goto_rule(rule) and evaluate_goto(rule['goto'], self.metadata, self.answer_store, this_location['group_instance']):
-                        is_meta_rule = SchemaHelper.is_goto_meta_rule(rule)
+                for rule in [r for r in block['routing_rules'] if SchemaHelper.is_goto_rule(r)]:
+                    should_goto = evaluate_goto(rule['goto'], self.metadata, self.answer_store, this_location['group_instance'])
+
+                    if should_goto:
                         next_location = this_location.copy()
                         next_location['block_id'] = rule['goto']['id']
 
                         next_block_index = self._block_index_for_location(blocks, next_location)
+                        next_precedes_current = next_block_index is not None and block_index > next_block_index
```

Comment:
I agree, but I think that it would read better using `next_block_index < block_index`